projects/sandbox/train/train/data_structures.py

```diff
@@ -118,8 +125,12 @@ def forward(self, X: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
             max_center_offset=self.trigger_offset,
             coincident=True,
         )
+
+        waveforms, indices = self.channel_swapper(waveforms)
         X[mask] += waveforms
 
+        # make targets negative if they have had a channel swapped
+        mask = mask[np.where(mask)[0]][indices] = 0
```

Comment:
Can you explain this line? I'm having trouble parsing it.

Reply:
The first `mask = ` is a typeo. Would that clarify any confusion?

Reply:
Ah, that helps. The shapes of everything is breaking my brain - why the `[0]`?

Reply:
Got it, okay, cool

Reply:
But in short, `mask` is a boolean array which refers to samples of the background where we will inject waveforms. Due to the channel swapping, some subset waveforms have a channel swapped, and therefore need to be labeled as noise. `Indices` indexes the subset of mask where mask is `True`. These are waveforms that have been swapped, and thus their label should be set to 0.